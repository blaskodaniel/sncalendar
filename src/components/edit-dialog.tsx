import Dialog, { DialogProps } from '@material-ui/core/Dialog'
import React from 'react'
import CalendarEvent from '../CalendarEvent-type'
import EditPropertiesDialogBody from './edit-dialog-body'

export const EditPropertiesDialog: React.FunctionComponent<{
  dialogProps: DialogProps
  content: CalendarEvent
}> = props => {
  return (
    <Dialog open={props.dialogProps.open} fullScreen aria-labelledby="event-dialog">
      <EditPropertiesDialogBody contentId={props.content.Id} dialogProps={props.dialogProps} />
    </Dialog>
  )
}
