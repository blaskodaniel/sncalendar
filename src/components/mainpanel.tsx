import React, { useEffect, useState } from 'react'
import { ODataCollectionResponse } from '@sensenet/client-core'
import moment from 'moment'
import sortby from 'lodash.sortby'
import { createStyles, makeStyles } from '@material-ui/styles'
import { List, ListItem, ListItemAvatar } from '@material-ui/core'
import { v1 } from 'uuid'
import { Query } from '@sensenet/query'
import CalendarEvent from '../CalendarEvent-type'
import { useRepository } from '../hooks/use-repository'
import EventComponent from './event'

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      width: '100%',
    },
    dayAvatar: {
      alignSelf: 'flex-start',
      paddingTop: '8px',
    },
    dayname: {
      display: 'block',
      textAlign: 'center',
    },
    daynumber: {
      display: 'block',
      textAlign: 'center',
      fontSize: '20px',
      fontWeight: 400,
    },
    nopadding: {
      padding: '0',
    },
  }),
)

export interface GroupdByAny {
  date: any
  event: CalendarEvent[]
  id: string
}

/**
 * Main component
 */
const MainPanel: React.FunctionComponent = () => {
  const classes = useStyles()
  const repo = useRepository()
  const [data, setData] = useState<GroupdByAny[]>([])

  const groupByDay = function(xs: CalendarEvent[], key: keyof Pick<CalendarEvent, 'StartDate'>) {
    const resultArray: GroupdByAny[] = []
    xs.forEach(event => {
      const findevent = resultArray.find(
        c =>
          moment(new Date(c.date)).format('YYYY-MM-DD') ===
          moment(new Date(event[key] !== undefined ? (event[key] as string) : '')).format('YYYY-MM-DD'),
      )
      if (findevent) {
        findevent.event.push(event)
        const valami = sortby(findevent.event, 'AllDay', ['asc'])
        console.log('sort: ', valami)
      } else {
        if (event[key]) {
          resultArray.push({
            id: v1(),
            date: event[key],
            event: [event],
          })
        }
      }
    })
    return resultArray
  }

  useEffect(() => {
    const loadCalendar = async () => {
      const result: ODataCollectionResponse<CalendarEvent> = await repo.loadCollection({
        path: `/Root/Content/IT/Calendar`,
        oDataOptions: {
          select: [
            'DisplayName',
            'Description',
            'CreationDate',
            'CreatedBy',
            'ModifiedBy',
            'ModificationDate',
            'Icon',
            'Type',
            'Id',
            'Path',
            'Name',
            'Size',
            'Location',
            'StartDate',
            'EndDate',
            'Lead',
            'AllDay',
            'EventUrl',
            'OwnerEmail',
          ] as any,
          query: new Query(q =>
            q.greatherThan('StartDate', '2019-01-01').and.lessThan('StartDate', '2019.12.31'),
          ).toString(),
          orderby: [['StartDate', 'asc']],
          expand: ['CreatedBy', 'ModifiedBy'],
        },
      })

      const groupedby = groupByDay(result.d.results, 'StartDate')
      console.log(groupedby)
      setData(groupedby)
    }

    // Load calendar datas from Repository
    loadCalendar()
  }, [repo])

  return (
    <>
      {data.map(element => {
        return (
          <List key={element.id} className={classes.root}>
            <ListItem className={classes.nopadding}>
              <ListItemAvatar className={classes.dayAvatar}>
                <div>
                  <span className={classes.dayname}>{moment(new Date(element.date)).format('ddd')}</span>
                  <span className={classes.daynumber}>{moment(new Date(element.date)).format('D')}</span>
                </div>
              </ListItemAvatar>
              <EventComponent event={element.event} />
            </ListItem>
          </List>
        )
      })}
    </>
  )
}

export default MainPanel
