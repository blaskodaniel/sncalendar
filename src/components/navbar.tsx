import React from 'react'
import { AppBar, Button, Toolbar, Tooltip, Typography } from '@material-ui/core'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import { useRepository } from '../hooks/use-repository'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
    },
  }),
)
/**
 * Navbar component
 */
export const NavBarComponent: React.FunctionComponent<{ month: string }> = props => {
  const repo = useRepository()
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <AppBar>
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            {props.month}
          </Typography>
          <Tooltip title="Return to the Login screen and select another repository">
            <Button color="inherit" onClick={() => repo.authentication.logout()}>
              Log out
            </Button>
          </Tooltip>
        </Toolbar>
      </AppBar>
    </div>
  )
}
