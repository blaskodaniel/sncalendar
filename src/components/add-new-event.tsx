import React, { useContext } from 'react'
import { createStyles, Fab, makeStyles, Theme } from '@material-ui/core'
import AddIcon from '@material-ui/icons/Add'
import { SharedContext } from '../context/shared-context'

/**
 * Style for component with Material UI
 */
const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    fabAdd: {
      position: 'fixed',
      bottom: theme.spacing(2),
      right: theme.spacing(2),
    },
    extendedIcon: {
      marginRight: theme.spacing(1),
    },
  }),
)

const AddNewEvent: React.FunctionComponent = () => {
  const classes = useStyles()
  const sharedcontext = useContext(SharedContext)

  return (
    <Fab
      color="secondary"
      variant="extended"
      onClick={() => {
        // New event
        sharedcontext.setOpennewmodal(true)
      }}
      aria-label="Add new"
      className={classes.fabAdd}>
      <AddIcon className={classes.extendedIcon} />
      Add new
    </Fab>
  )
}

export default AddNewEvent
