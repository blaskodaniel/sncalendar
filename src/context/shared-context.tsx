import React, { createContext, Dispatch, useState } from 'react'
import { GenericContent } from '@sensenet/default-content-types'
import { EditPropertiesDialog } from '../components/edit-dialog'
import CalendarEvent from '../CalendarEvent-type'
import { DialogComponent } from '../components/view-dialog'

export const SharedContext = createContext<{
  openeditmodal: boolean
  setOpeneditmodal: Dispatch<React.SetStateAction<boolean>>
  opendisplaymodal: boolean
  setOpendisplaymodal: Dispatch<React.SetStateAction<boolean>>
  setEvent: Dispatch<React.SetStateAction<GenericContent | undefined>>
}>(null as any)

const SharedProvider: React.FunctionComponent<any> = props => {
  const [openeditmodal, setOpeneditmodal] = useState(false)
  const [opendisplaymodal, setOpendisplaymodal] = useState(false)
  const [event, setEvent] = useState<CalendarEvent>(null as any)

  return (
    <>
      <SharedContext.Provider
        value={{ openeditmodal, setOpeneditmodal, setEvent, opendisplaymodal, setOpendisplaymodal }}>
        {props.children}
        {openeditmodal && event != null ? (
          <EditPropertiesDialog
            content={event}
            dialogProps={{
              open: openeditmodal,
              onClose: () => setOpeneditmodal(false),
              keepMounted: false,
            }}
          />
        ) : null}
        {opendisplaymodal && event != null ? (
          <DialogComponent open={opendisplaymodal} content={event} onClose={() => setOpendisplaymodal(false)} />
        ) : null}
      </SharedContext.Provider>
    </>
  )
}
export default SharedProvider
