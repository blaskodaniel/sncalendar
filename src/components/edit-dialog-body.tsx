import React, { useContext } from 'react'
import { DialogContent, DialogTitle } from '@material-ui/core'
import { EditView } from '@sensenet/controls-react'
import { DialogProps } from '@material-ui/core/Dialog'
import { ConstantContent } from '@sensenet/client-core'
import { ContentType } from '@sensenet/default-content-types'
import { useRepository } from '../hooks/use-repository'
import { useSelectionService } from '../hooks/use-selection-service'
import { CurrentContentContext, CurrentContentProvider } from '../context/current-context'
import CalendarEvent from '../CalendarEvent-type'
import { SharedContext } from '../context/shared-context'

const EditPropertiesDialogBody: React.FunctionComponent<{
  contentId: number
  dialogProps: DialogProps
}> = props => {
  const selectionService = useSelectionService()
  const repo = useRepository()
  const sharedcontext = useContext(SharedContext)

  const onSubmit = async (id: number, content: CalendarEvent) => {
    try {
      const response = await repo.patch({
        idOrPath: id,
        content,
        oDataOptions: {
          select: 'all' as any,
        },
      })
      sharedcontext.setEvent(response.d) // Refresh the view dialog window
      sharedcontext.setRefreshcalendar(!sharedcontext.refreshcalendar) // Refresh the main calendar list
      props.dialogProps.onClose && props.dialogProps.onClose(null as any, 'backdropClick')
    } catch (error) {
      console.log('Error: ', error)
    }
  }

  return (
    <CurrentContentProvider
      idOrPath={props.contentId}
      onContentLoaded={(c: CalendarEvent) => selectionService.activeContent.setValue(c)}
      oDataOptions={{ select: 'all' }}>
      <CurrentContentContext.Consumer>
        {(content: CalendarEvent) =>
          content.Id !== ConstantContent.PORTAL_ROOT.Id && (
            <>
              <DialogTitle>Edit</DialogTitle>
              <DialogContent>
                <EditView
                  content={content as ContentType}
                  repository={repo}
                  handleCancel={() =>
                    props.dialogProps.onClose && props.dialogProps.onClose(null as any, 'backdropClick')
                  }
                  onSubmit={onSubmit}
                />
              </DialogContent>
            </>
          )
        }
      </CurrentContentContext.Consumer>
    </CurrentContentProvider>
  )
}

export default EditPropertiesDialogBody
