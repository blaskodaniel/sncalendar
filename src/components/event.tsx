import React, { useContext } from 'react'
import moment from 'moment'
import { Avatar, createStyles, List, ListItem, ListItemAvatar, ListItemText, makeStyles } from '@material-ui/core'
import { User } from '@sensenet/default-content-types'
import CalendarEvent from '../CalendarEvent-type'
import defavatar from '../assets/avatar-default.png'
import { useRepository } from '../hooks/use-repository'
import { SharedContext } from '../context/shared-context'

const useStyles = makeStyles(() =>
  createStyles({
    parentlistelement: {
      display: 'flex',
      flexDirection: 'row-reverse',
      minWidth: '350px',
      border: '1px solid rgba(0, 0, 0, 0.12)',
      borderRadius: '2%',
      marginBottom: '7px',
      '&:hover': {
        backgroundColor: '#bad892',
        cursor: 'pointer',
      },
    },
    alldayevent: {
      backgroundColor: '#d3daff',
    },
    simpleevent: {
      backgroundColor: 'azure',
    },
  }),
)

export interface EventComponentProps {
  event: CalendarEvent[]
}

const EventComponent: React.FunctionComponent<EventComponentProps> = props => {
  const classes = useStyles()
  const repo = useRepository()
  const sharedcontext = useContext(SharedContext)
  const timeAndLocationpart = (event: CalendarEvent) => {
    const start = moment(new Date(event.StartDate as string)).format('HH:mm')
    const end = moment(new Date(event.EndDate as string)).format('HH:mm')
    const isAllday = event.AllDay
    const location = event.Location
    const timeDisplay = isAllday ? 'all day' : ` ${start}-${end}`
    return `${timeDisplay} at ${location}`
  }

  return (
    <List>
      {props.event.map(event => {
        return (
          <div key={event.Id}>
            <ListItem
              onClick={() => {
                sharedcontext.setEvent(event)
                sharedcontext.setOpendisplaymodal(true)
              }}
              className={`${classes.parentlistelement} ${event.AllDay ? classes.alldayevent : classes.simpleevent}`}>
              <ListItemAvatar>
                <Avatar
                  alt="Avatar"
                  src={
                    (event.CreatedBy as User).Avatar!.Url === ''
                      ? defavatar
                      : repo.configuration.repositoryUrl + (event.CreatedBy as User).Avatar!.Url
                  }
                />
              </ListItemAvatar>
              <ListItemText primary={event.DisplayName} secondary={timeAndLocationpart(event)} />
            </ListItem>
          </div>
        )
      })}
    </List>
  )
}

export default EventComponent
