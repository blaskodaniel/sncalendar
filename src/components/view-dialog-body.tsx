import React, { useContext, useState } from 'react'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import moment from 'moment'
import striptags from 'striptags'
import {
  Avatar,
  Card,
  CardContent,
  Grid,
  IconButton,
  List,
  ListItem,
  ListItemAvatar,
  ListItemText,
  Typography,
} from '@material-ui/core'
import RoomIcon from '@material-ui/icons/Room'
import WatchIcon from '@material-ui/icons/Watch'
import NotesIcon from '@material-ui/icons/Notes'
import EditOutlined from '@material-ui/icons/EditOutlined'
import DeleteOutlined from '@material-ui/icons/DeleteOutlined'
import CalendarEvent from '../CalendarEvent-type'
import { SharedContext } from '../context/shared-context'
import { RepositoryContext } from '../context/repository-provider'
import { DialogComponent } from './simple-dialog'

export interface EditDialogBodyDialogProps {
  content: CalendarEvent
  dialogClose: (value: boolean) => void
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    listitemroot: {
      width: '100%',
      maxWidth: 450,
      padding: 0,
      backgroundColor: theme.palette.background.paper,
    },
    ml15: {
      marginLeft: '15px',
    },
    card: {
      boxShadow: 'none',
    },
    media: {
      height: 0,
      paddingTop: '56.25%', // 16:9
    },
    expand: {
      transform: 'rotate(0deg)',
      marginLeft: 'auto',
      transition: theme.transitions.create('transform', {
        duration: theme.transitions.duration.shortest,
      }),
    },
    expandOpen: {
      transform: 'rotate(180deg)',
    },
    avatar: {
      backgroundColor: '#4d8ead',
    },
    icon: {
      fontSize: '2rem',
    },
    button: {
      padding: '4px',
    },
    notificationmargin: {
      margin: theme.spacing(1),
    },
  }),
)

export const ViewDialogBody: React.FunctionComponent<EditDialogBodyDialogProps> = props => {
  const classes = useStyles()
  const repo = useContext(RepositoryContext)
  const sharedcontext = useContext(SharedContext)
  const [openmodal, setOpenmodal] = useState(false)

  const dateTimeDisplay = (event: CalendarEvent) => {
    if (event.AllDay) {
      return 'All day'
    }
    if (moment(event.StartDate).isSame(event.EndDate, 'day')) {
      return `${moment(event.StartDate).format('dddd, MMM-DD HH:mm')} - ${moment(event.EndDate).format('HH:mm')}`
    } else {
      return `${moment(event.StartDate).format('dddd, MMM-DD HH:mm')} - ${moment(event.EndDate).format(
        'dddd, MMM-DD HH:mm',
      )}`
    }
  }

  const deleteEvent = async (userevent: boolean, event: CalendarEvent) => {
    if (userevent) {
      await repo.delete({ idOrPath: event.Path, permanent: true })
      sharedcontext.setRefreshcalendar(!sharedcontext.refreshcalendar)
      sharedcontext.setOpendisplaymodal(false)
      sharedcontext.setOpennoti(true)
    } else {
      setOpenmodal(false)
    }
  }

  return (
    <>
      <Card className={classes.card}>
        <CardContent>
          <Grid container direction="row" justify="flex-end">
            <Grid item>
              <IconButton
                aria-label="remove"
                className={classes.button}
                onClick={() => {
                  setOpenmodal(true)
                }}>
                <DeleteOutlined className={classes.icon} />
              </IconButton>
              <IconButton
                aria-label="edit"
                className={classes.button}
                onClick={() => {
                  sharedcontext.setEvent(props.content)
                  sharedcontext.setOpeneditmodal(true)
                }}>
                <EditOutlined className={classes.icon} />
              </IconButton>
            </Grid>
          </Grid>
          <Grid container direction="column" justify="center">
            <Grid item>
              <Typography variant="h5" component="h2" className={classes.ml15}>
                {props.content.DisplayName}
              </Typography>
            </Grid>
            <Grid item>
              <List className={classes.listitemroot}>
                <ListItem>
                  <ListItemAvatar>
                    <Avatar>
                      <WatchIcon />
                    </Avatar>
                  </ListItemAvatar>
                  <ListItemText primary="Date" secondary={dateTimeDisplay(props.content)} />
                </ListItem>
              </List>
            </Grid>
            <Grid item>
              <List className={classes.listitemroot}>
                <ListItem>
                  <ListItemAvatar>
                    <Avatar>
                      <RoomIcon />
                    </Avatar>
                  </ListItemAvatar>
                  <ListItemText primary="Location" secondary={props.content.Location} />
                </ListItem>
              </List>
            </Grid>
            <Grid item>
              <List className={classes.listitemroot}>
                <ListItem>
                  <ListItemAvatar>
                    <Avatar>
                      <NotesIcon />
                    </Avatar>
                  </ListItemAvatar>
                  <ListItemText
                    primary="Description"
                    secondary={props.content.Description != undefined ? striptags(props.content.Description) : ''}
                  />
                </ListItem>
              </List>
            </Grid>
          </Grid>
        </CardContent>
      </Card>
      <DialogComponent
        open={openmodal}
        title={props.content.DisplayName ? props.content.DisplayName : ''}
        onClose={userevent => {
          deleteEvent(userevent, props.content)
        }}
      />
    </>
  )
}
