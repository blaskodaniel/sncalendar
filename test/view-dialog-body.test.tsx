import { mount, shallow } from 'enzyme'
import React from 'react'
import IconButton from '@material-ui/core/IconButton'
import { act } from 'react-dom/test-utils'
import { Repository } from '@sensenet/client-core'
import { Button } from '@material-ui/core'
import { RepositoryContext } from '../src/context/repository-provider'
import { SharedContext } from '../src/context/shared-context'
import { ViewDialogBody } from '../src/components/view-dialog-body'
import { CalendarTestEvent, CalendarTestEventAllDay } from './mocks/test-objects'

jest.mock('react-quill')

describe('ViewDialogBody', () => {
  const content = CalendarTestEvent
  const contentAllDay = CalendarTestEventAllDay
  const testprops = {
    content,
    dialogClose: true,
  }
  const testpropsAllday = {
    content: contentAllDay,
    dialogClose: true,
  }
  const shareobject = {
    openeditmodal: true,
    setOpeneditmodal: jest.fn(),
    setEvent: jest.fn(),
    opendisplaymodal: false,
    setOpendisplaymodal: jest.fn(),
    event: content,
    refreshcalendar: true,
    setRefreshcalendar: jest.fn(),
    opennewmodal: true,
    setOpennewmodal: jest.fn(),
    setOpennoti: jest.fn(),
  }
  const shareobjectAllDay = {
    openeditmodal: true,
    setOpeneditmodal: jest.fn(),
    setEvent: jest.fn(),
    opendisplaymodal: false,
    setOpendisplaymodal: jest.fn(),
    event: contentAllDay,
    refreshcalendar: true,
    setRefreshcalendar: jest.fn(),
    opennewmodal: true,
    setOpennewmodal: jest.fn(),
    setOpennoti: jest.fn(),
  }

  it('ViewDialogBody snapshot test', () => {
    const wrapper = shallow(<ViewDialogBody {...(testprops as any)} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Open dialog and remove event test', async () => {
    const repo = new Repository()
    repo.load = function load() {
      return Promise.resolve({ d: content } as any)
    }
    repo.delete = jest.fn()
    let wrapper: any
    await act(async () => {
      wrapper = mount(
        <RepositoryContext.Provider value={repo as any}>
          <SharedContext.Provider value={shareobject as any}>
            <ViewDialogBody {...(testprops as any)} />
          </SharedContext.Provider>
        </RepositoryContext.Provider>,
      )
    })
    wrapper
      .update()
      .find(IconButton)
      .first()
      .simulate('click')
    expect(wrapper.find('div#simple-dialog-title').text()).toEqual('Are you sure you want to delete it?')

    await act(async () => {
      wrapper
        .find(Button)
        .first()
        .simulate('click')
    })

    expect(shareobject.setRefreshcalendar).toBeCalledWith(!shareobject.refreshcalendar)
    expect(shareobject.setOpendisplaymodal).toBeCalledWith(false)
    expect(shareobject.setOpennoti).toBeCalledWith(true)
  })

  it('All Day event test', async () => {
    const repo = new Repository()
    repo.load = function load() {
      return Promise.resolve({ d: content } as any)
    }
    let wrapper: any
    await act(async () => {
      wrapper = mount(
        <RepositoryContext.Provider value={repo as any}>
          <SharedContext.Provider value={shareobjectAllDay as any}>
            <ViewDialogBody {...(testpropsAllday as any)} />
          </SharedContext.Provider>
        </RepositoryContext.Provider>,
      )
    })

    expect(wrapper.find('div.MuiCardContent-root').text()).toContain('All day')
  })
})
