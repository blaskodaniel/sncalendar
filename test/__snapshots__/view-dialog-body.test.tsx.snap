// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewDialogBody ViewDialogBody snapshot test 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-3"
  >
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        direction="row"
        justify="flex-end"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(IconButton))
            aria-label="remove"
            className="makeStyles-button-9"
            onClick={[Function]}
          >
            <DeleteOutlinedIcon
              className="makeStyles-icon-8"
            />
          </WithStyles(ForwardRef(IconButton))>
          <WithStyles(ForwardRef(IconButton))
            aria-label="edit"
            className="makeStyles-button-9"
            onClick={[Function]}
          >
            <EditOutlinedIcon
              className="makeStyles-icon-8"
            />
          </WithStyles(ForwardRef(IconButton))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        direction="column"
        justify="center"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-ml15-2"
            component="h2"
            variant="h5"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(List))
            className="makeStyles-listitemroot-1"
          >
            <WithStyles(ForwardRef(ListItem))>
              <WithStyles(ForwardRef(ListItemAvatar))>
                <WithStyles(ForwardRef(Avatar))>
                  <WatchIcon />
                </WithStyles(ForwardRef(Avatar))>
              </WithStyles(ForwardRef(ListItemAvatar))>
              <WithStyles(ForwardRef(ListItemText))
                primary="Date"
                secondary="Tuesday, Jan-01 13:00 - 14:00"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(List))
            className="makeStyles-listitemroot-1"
          >
            <WithStyles(ForwardRef(ListItem))>
              <WithStyles(ForwardRef(ListItemAvatar))>
                <WithStyles(ForwardRef(Avatar))>
                  <RoomIcon />
                </WithStyles(ForwardRef(Avatar))>
              </WithStyles(ForwardRef(ListItemAvatar))>
              <WithStyles(ForwardRef(ListItemText))
                primary="Location"
                secondary="Budapest"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(List))
            className="makeStyles-listitemroot-1"
          >
            <WithStyles(ForwardRef(ListItem))>
              <WithStyles(ForwardRef(ListItemAvatar))>
                <WithStyles(ForwardRef(Avatar))>
                  <NotesIcon />
                </WithStyles(ForwardRef(Avatar))>
              </WithStyles(ForwardRef(ListItemAvatar))>
              <WithStyles(ForwardRef(ListItemText))
                primary="Description"
                secondary="Event description"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
  <Component
    onClose={[Function]}
    open={false}
    title=""
  />
</Fragment>
`;
